File: CareConnect/CareConnect.Models/Requests/UserInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class UserInsertRequest
    {
        public string FirstName { get; set; } = null!;

        public string LastName { get; set; } = null!;

        public string? Email { get; set; }

        public string? PhoneNumber { get; set; }

        public string Username { get; set; } = null!;

        public string Password { get; set; } = null!;

        public string ConfirmationPassword { get; set; } = null!;

        [JsonIgnore]
        public bool Status { get; set; } = true; 

        public DateTime? BirthDate { get; set; }

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now; 

        public string Gender { get; set; } = null!;

        public string? Address { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/requests/user_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_insert_request.g.dart';

@JsonSerializable()
class UserInsertRequest {
  final String firstName;
  final String lastName;
  final String? email;
  final String? phoneNumber;
  final String username;
  final String password;
  final String confirmationPassword;
  final DateTime? birthDate;
  final String gender;
  final String? address;

  UserInsertRequest({
    required this.firstName,
    required this.lastName,
    this.email,
    this.phoneNumber,
    required this.username,
    required this.password,
    required this.confirmationPassword,
    this.birthDate,
    required this.gender,
    this.address,
  });

  /// Connect the generated [_$EmployeeFromJson] function to the `fromJson`
  /// factory.
  factory UserInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserInsertRequestFromJson(json);

  /// Connect the generated [_$UserToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserInsertRequestToJson(this);
}
